### ISSUE TITLE
Can't reference a specific function overload

### ISSUE THREAD

**Issue by porokoro** [Kotlin/dokka#80]
Consider this scenario:
I have two or more methods with the same name but different parameters, just simple function overloads. Then I can't reference one of them explicitly in the documentation.

For example:
```kotlin
class Sample {
    /**
     * If calling this, don't call [fancy] or [fancy] before [build].
     */
    fun fancy(value: String) { ... }
    fun fancy(value: Int) { ... }
    fun fancy(value: String, extra: String) { ... }
    fun build(): Something { ... }
}
```

Something like `[fancy(Int)]` and `[fancy(String, String)]` would be nice in the scenario, to separate the reference by parameter types and thus being able to reference one specific overload.

**Comment by alexfacciorusso** [comment]
Same problem here.

**Comment by fabienmoritz** [comment]
It is never too late to answer a good suggestion ! :sweat_smile:

**Comment by Sxtanna** [comment]
This should 100% be supported, this is essential...

**Comment by NoHarmDan** [comment]
How is this still not supported?! The suggested solution would look perfectly fine.

**Comment by alllex** [comment]
Moreover, this missing feature may make documentation browsing a bit misleading. It may be possible to dodge overloading with default parameters in Kotlin, but impossible when operating with JDK classes. 

Consider the case when you mention Java method which has several overloads in your Kotlin doc of `foo`. When you open a quick documentation for `foo` the Java method mention is shown as a link clicking on which may bring you to the documentation of an **incorrect** overload.

**Comment by chrisjenx** [comment]
This still isn't supported?

**Comment by MikhailShumenko** [comment]
Is it fixed?

**Comment by fo2rist** [comment]
Looks like it's not.
Any news from JetBrains team?

**Comment by zach-klippenstein** [comment]
It may not be required for HTML outputs, but when linking to code in the IDE it is extremely frustrating to not be able to link to a specific overload. E.g. Clicking a link to `Job.cancel` will take you to the deprecated `cancel()` function. Clicking a link to `kotlinx.coroutines.cancel` will take you to `CoroutineContext.cancel`, and there's no way to link to `CoroutineScope.cancel`.

**Comment by christopherfrieler** [comment]
I just came across this limitation and would definitely appreciate to see this fixed for the same reason: it is really annoying to have these "broken" links in my IDE.

At the same time I'm not sure if this is something to be done in dokka in the first place. The Kotlin Spec doesn't say anything about how links into Kotlin code are specified or resolved (see [Documentation comments](https://kotlin.github.io/kotlin-spec/#documentation-comments)). Instead there's a big TODO.

I think the Spec should specify this first. Then (or maybe simultaneously) tools like dokka or any IDE can implement that. I assume IntelliJ/Android Studio doesn't use dokka internally anyway to resolve those links, so just implementing that in dokka wouldn't even help in my IDE.

**Comment by DavidSonoda** [comment]
> It is never too late to answer a good suggestion ! 😅

Strongly agreed! It's confusing to see those broken links.

**Comment by cgruber** [comment]
I've now renamed methods to avoid this issue, and not actually useful renames. Please consider supporting function overload targeting, especially since not all overloads are the result of defaults, but may take very different types (but do the same thing, and thus be usefully named the same.

**Comment by cutiko** [comment]
I think this should be supported to reference functions from the Java Android API. My use case is when referencing using the `@see` annotation

**Comment by vincentfree** [comment]
Same problem, this looks like something that just should be there. I'm trying to document some usage of the jdk Cipher's doFinal method, im trying to express the usage of `.doFinal(ByteArray)` but it defaults too `.doFinal()` this isn't what's happening so the documentation isn't documenting the intent this way

**Comment by vad-zuev** [comment]
still no updates on this? I can just repeat what others said: this is really frustrating ><

**Comment by kamildoleglo** [comment]
Sorry, still nothing from our side as this is a change in KDoc that must be done by the IntelliJ team, here's the relevant ticket: https://youtrack.jetbrains.com/issue/KT-15984

**Comment by akamuza** [comment]
Still an issue in 2021

**Comment by ichenhe** [comment]
I have to give the function a different name. It's troublesome.

**Comment by Ahmed-Mostafa-Ghanayem** [comment]
This issue will celebrate 6th birthday soon and still no updates.

**Comment by IgnatBeresnev** [comment]
Hi, yeah :) Unfortunately, this problem spans beyond dokka (kdoc spec, probably some compiler/ide work as well), so it's not as easy to fix, but it's definitely on our radar, we'll keep you posted if anything update-worthy comes up.

**Comment by vmishenev** [comment]
This is actual for a function and a property with the same name:
```kt
val name = ""
fun name() = ""
```

**Comment by HalfLegend** [comment]
Still an issue in 2023

**Comment by rexmtorres** [comment]
> Still an issue in 2023

And still in 2024

**Comment by wallind** [comment]
Adding my voice as someone who just went to do this, and found I could not.

Also, to add a laugh, JetBrains AI Asssitant lied to me and told me I could 🙃 ...lol - [ai-telling-lies-about-kdoc.txt](https://github.com/Kotlin/dokka/files/14876023/ai-telling-lies-about-kdoc.txt)

**Comment by aaprokofyev** [comment]
+1

**Comment by damiankaplonspa** [comment]
+1

**Comment by Chuckame** [comment]
It seems that the radar has some issues! Still the same in 2025

**Comment by 5GameMaker** [comment]
It'll be also nice to have `[var fieldName]`/`[val fieldName]` syntax to explicitly reference a var. (`[val fieldName set(..)]` for setter and `[val fieldName get()]` for getter)

Also still an issue in 2026.

**Comment by CLOVIS-AI** [comment]
It looks like progress is being made → https://github.com/Kotlin/dokka/issues/4425

**Comment by whyoleg** [comment]
> It looks like progress is being made → https://github.com/Kotlin/dokka/issues/4425

No, it's not. New KDoc link resolution is about something else. :)

---

Though of course we know about the issue with overloads and its importance, it's on our radar, and we want to design a solution for it in the foreseeable future. But, right now, no dates, no promises. :)

**Comment by CLOVIS-AI** [comment]
Ah, sorry 🙏

**Comment by ununhexium** [comment]
Happy 10 year anniversary 🎂

**Comment by Chuckame** [comment]
The funny part is that it's the most upvoted issue **all-time** on the repository.

<img width="1126" height="383" alt="Image" src="https://github.com/user-attachments/assets/d7a44bd0-0463-4b63-8b72-a929551b9875" />

It would be interesting to list the impacts, and start working on the spec proposal... But we need to know if it's worth the work to be considered by the maintainers' team
